dev/prepare_jvm_release.py

```diff
@@ -47,57 +50,76 @@ def cd(path):
     finally:
         os.chdir(cwd)
 
+
 def run(command, **kwargs):
     print(command)
     subprocess.check_call(command, shell=True, **kwargs)
 
+
 def get_current_git_tag():
     out = subprocess.check_output(["git", "tag", "--points-at", "HEAD"])
     return out.decode().split("\n")[0]
 
+
 def get_current_commit_hash():
     out = subprocess.check_output(["git", "rev-parse", "HEAD"])
     return out.decode().split("\n")[0]
 
+
 def get_current_git_branch():
     out = subprocess.check_output(["git", "log", "-n", "1", "--pretty=%d", "HEAD"])
     m = re.search(r"release_[0-9\.]+", out.decode())
     if not m:
         raise ValueError("Expected branch name of form release_xxx")
     return m.group(0)
 
+
 def retrieve(url, filename=None):
     print(f"{url} -> {filename}")
     return urlretrieve(url, filename)
 
+
 def main():
     parser = argparse.ArgumentParser()
-    parser.add_argument("--release-version", type=str, required=True,
-                        help="Version of the release being prepared")
+    parser.add_argument(
+        "--release-version",
+        type=str,
+        required=True,
+        help="Version of the release being prepared",
+    )
     args = parser.parse_args()
 
-    if sys.platform != "darwin" or platform.machine() != "x86_64":
-        raise NotImplementedError("Please run this script using an Intel Mac")
+    if sys.platform != "darwin" or platform.machine() != "arm64":
```

Comment:
What is the reason the intel macs are excluded?

Reply:
The native binary for Intel Mac is built by the CI already. So I need to run the prep script on M1 mac to build the binary for M1. You can look at the content of the JAR to see that there are multiple libxgboost4j.dylib.